src/annmas/scsplit/command.py

```diff
@@ -0,0 +1,300 @@
+import logging
+import sys
+import itertools
+
+import time
+
+import click
+import click_log
+import tqdm
+
+import pysam
+import multiprocessing as mp
+
+from inspect import getframeinfo, currentframe, getdoc
+
+from ..utils.model import array_element_structure
+
+from ..annotate.command import SegmentInfo
+from ..annotate.command import SEGMENTS_TAG
+from ..annotate.command import _get_segments
+
+from ..utils.constants import __SENTINEL_VALUE
+
+from ..meta import VERSION
+
+logging.basicConfig(stream=sys.stderr)
+logger = logging.getLogger("scsplit")
+click_log.basic_config(logger)
+
+
+__DEFAULT_DUMMY_CBC = "CTGCCTAACCTGATCC"
+__DEFAULT_OUT_BASE_NAME = "scsplit"
+__DEFAULT_UMI_LENGTH = 10
+
+
+__OUT_READ_FILE_SUFFIX = "_mates"
+__OUT_WHITELIST_FILE_SUFFIX = "_whitelist.txt"
+
+
+@click.command(name=logger.name)
+@click_log.simple_verbosity_option(logger)
+@click.option(
+    "-t",
+    "--threads",
+    type=int,
+    default=mp.cpu_count() - 1,
+    show_default=True,
+    help="number of threads to use (0 for all)",
+)
+@click.option(
+    "-o",
+    "--output-base-name",
+    default=__DEFAULT_OUT_BASE_NAME,
+    type=str,
+    help=f"base name for output files [default: {__DEFAULT_OUT_BASE_NAME}]",
+)
+@click.option(
+    "-c",
+    "--cell-barcode",
+    default=__DEFAULT_DUMMY_CBC,
+    type=str,
+    help=f"dummy cell barcode to use for the dataset [default: {__DEFAULT_DUMMY_CBC}, "
+         f"length: {len(__DEFAULT_DUMMY_CBC)}]",
+)
+@click.option(
+    "-u",
+    "--umi-length",
+    default=__DEFAULT_UMI_LENGTH,
+    type=int,
+    show_default=True,
+    help=f"length of the UMI from this library",
+)
+@click.argument(
+    "input-bam",
+    default="-" if not sys.stdin.isatty() else None,
+    type=click.File("rb"),
+)
+def main(threads, output_base_name, cell_barcode, umi_length, input_bam):
+    f"""Create a set of files suitable for use in `alevin` for single-cell analysis.
```

Comment:
Ah.  Yeah - this is essentially because of the f-string, but doesn't work even if you concatenate strings directly.  Click needs to have the docstrings finalized by the time it parses them, which seems to be very early.

I updated the docstring to be static.